src/spark_rapids_ml/utils.py

```diff
@@ -122,6 +122,26 @@ def build(cls, partition_rows: List[int], total_cols: int) -> "PartitionDescript
         return cls(total_rows, total_cols, rank, parts_rank_size)
 
 
+def _concat_and_free(np_array_list: List[np.ndarray]) -> np.ndarray:
+    """
+    concatenates a list of compatible numpy arrays into a F ordered output array,
+    in a memory efficient way.
+    Note: frees list elements so do not reuse after calling.
+    """
+    rows = sum(arr.shape[0] for arr in np_array_list)
+    if len(np_array_list[0].shape) > 1:
+        cols = np_array_list[0].shape[1]
+        concat_shape: Tuple[int, ...] = (rows, cols)
```

Comment:
Not sure if this was intended to support more than two dimensions (because of the ellipsis), but `cols` as-written is only a single dimension, e.g. did you intend something more like this?
```
# 2+ dimensions
cols = np_array_list[0].shape[1:]
concat_shape: Tuple[int, ...] = (rows,) + cols
```

Reply:
It declares concat_shape as a variable length tuple type due to the else branch assigning only a single component tuple.  mypy complained.